crates/swc_isograph_plugin/transform/src/lib.rs

```diff
@@ -0,0 +1,406 @@
+            span: unresolved_mark
+                .map(|m| DUMMY_SP.apply_mark(m))
+                .unwrap_or_default(),
+            sym: ident_name.into(),
+            optional: false,
+        })
+    }
+
+    fn build_require_expr_from_path(&self, path: &str, mark: Option<Mark>) -> Expr {
+        Expr::Member(MemberExpr {
+            span: DUMMY_SP,
+            obj: Box::new(Expr::Call(CallExpr {
+                span: DUMMY_SP,
+                callee: quote_ident!(
+                    mark.map(|m| DUMMY_SP.apply_mark(m)).unwrap_or(DUMMY_SP),
+                    "require"
+                )
+                .as_callee(),
+                args: vec![Lit::Str(Str {
+                    span: Default::default(),
+                    value: JsWord::from(path),
+                    raw: None,
+                })
+                .as_arg()],
+                type_args: None,
+            })),
+            prop: MemberProp::Ident(Ident {
+                sym: "default".into(),
+                span: DUMMY_SP,
+                optional: false,
+            }),
+        })
+    }
+
+    fn build_arrow_identity_expr(&self) -> Expr {
+        Expr::Arrow(ArrowExpr {
+            params: vec![Pat::Ident(Ident::new("x".into(), DUMMY_SP).into())],
+            body: Box::new(Ident::new("x".into(), DUMMY_SP).into()),
+            span: DUMMY_SP,
+            is_async: Default::default(),
+            is_generator: Default::default(),
+            return_type: Default::default(),
+            type_params: Default::default(),
+        })
+    }
+
+    fn path_for_artifact(
+        &self,
+        real_filepath: &Path,
+        config: &IsographProjectConfig,
+        root_dir: &Path,
+    ) -> Result<PathBuf, IsograthTransformError> {
+        let folder = PathBuf::from(real_filepath.parent().unwrap());
+        let cwd = PathBuf::from(root_dir);
+
+        let artifact_directory = cwd
+            .join(
+                config
+                    .artifact_directory
+                    .as_ref()
+                    .unwrap_or(&config.project_root),
+            )
+            .join(ISOGRAPH_FOLDER);
+        let artifact_directory = artifact_directory.as_path();
+
+        debug!("artifact_directory: {:#?}", artifact_directory);
+
+        let file_to_artifact_dir = &pathdiff::diff_paths(artifact_directory, folder)
+            .expect("Expected path to be diffable");
+
+        let mut file_to_artifact = PathBuf::from(format!(
+            "{}/{}/{}/{}.ts",
+            file_to_artifact_dir.display(),
+            self.field_type,
+            self.field_name,
+            self.artifact_type
+        ));
+
+        if cfg!(target_os = "windows") {
+            file_to_artifact = PathBuf::from(format!("{}", file_to_artifact.display()).replace("\\", "/"));
+        }
+        
+        if file_to_artifact.starts_with(ISOGRAPH_FOLDER) {
+            file_to_artifact = PathBuf::from(format!("./{}", file_to_artifact.display()));
+        }
+
+
```

Comment:
odd that this didn't complain, can you run `cargo fmt`? That will remove the space here. I have my vscode setup as follows

```
  "editor.formatOnSave": true,
```

and it formats correctly